### PR TITLE
Relation improvements

### PR DESCRIPTION
**Pull request by grafovdenis**
* **Breaking Change:** `EntityStateBuilder`'s `child` argument replaced with `builder`
* **Breaking Change:** `EntityStateBuilder`'s `errorBuilder` argument now passes `Widget Function(BuildContext, Exception)`.
If you'd like to get `data` from `error` please use `errorDataBuilder` instead.
* Add the ability to get an `error` in `EntityStateBuilder` from `errorChild`.
* Add `data` argument to `EntityStreamedState`.
* Update `rxdart` dependency to `0.27.0`.

### CODE REVIEW

File: packages/relation/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## 1.2.0
```
Comment (dkrutskikh) [inline packages/relation/CHANGELOG.md:3]: @grafovdenis please don't bump package version

File: packages/relation/lib/src/builder/entity_state_builder.dart [hunk 2]
```diff
@@ -60,8 +61,11 @@ class EntityStateBuilder<T> extends StatelessWidget {
   /// Loading child of builder
   final DataWidgetBuilder<T>? loadingBuilder;
 
+  /// Error child of builder with previous data
+  final DataWidgetErrorBuilder<T>? errorDataBuilder;
+
   /// Error child of builder
-  final DataWidgetErrorBuilder<T>? errorBuilder;
+  final ErrorWidgetBuilder? errorBuilder;
```
Comment (dkrutskikh) [inline packages/relation/lib/src/builder/entity_state_builder.dart:68]: @grafovdenis looks like breaking changes, is it true?

File: packages/relation/lib/src/builder/entity_state_builder.dart [hunk 3]
```diff
@@ -83,15 +87,23 @@ class EntityStateBuilder<T> extends StatelessWidget {
...
               streamData.data,
               streamData.error?.e != null
                   ? streamData.error!.e as Exception
                   : Exception(),
             );
           } else {
+            if (errorBuilder != null) {
+              return errorBuilder!(
+                context,
+                streamData.error?.e != null
```
Comment (dkrutskikh) [inline packages/relation/lib/src/builder/entity_state_builder.dart:102]: @grafovdenis please extract to variable

File: packages/relation/lib/src/builder/entity_state_builder.dart [hunk 4]
```diff
@@ -83,15 +87,23 @@ class EntityStateBuilder<T> extends StatelessWidget {
...
+          if (errorDataBuilder != null) {
+            return errorDataBuilder!(
               context,
               streamData.data,
               streamData.error?.e != null
                   ? streamData.error!.e as Exception
                   : Exception(),
             );
           } else {
+            if (errorBuilder != null) {
```
Comment (dkrutskikh) [inline packages/relation/lib/src/builder/entity_state_builder.dart:99]: @grafovdenis please cover by tests

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

**Review by arefimenko — APPROVED** [review]

File: packages/relation/lib/src/builder/entity_state_builder.dart [hunk 5]
```diff
@@ -83,19 +87,20 @@ class EntityStateBuilder<T> extends StatelessWidget {
...
-              streamData.data,
-              streamData.error?.e != null
-                  ? streamData.error!.e as Exception
-                  : Exception(),
-            );
+          final exception = streamData.error?.e != null
+              ? streamData.error!.e as Exception
+              : Exception();
+
+          if (errorDataBuilder != null) {
```
Comment (IlyaMax) [inline packages/relation/lib/src/builder/entity_state_builder.dart:94]: I think it'd better to add docs about priority of this builders and that errorChild has the lowest one.

File: packages/relation/lib/src/builder/entity_state_builder.dart [hunk 6]
```diff
@@ -83,19 +87,20 @@ class EntityStateBuilder<T> extends StatelessWidget {
...
-          if (errorBuilder != null) {
-            return errorBuilder!(
-              context,
-              streamData.data,
-              streamData.error?.e != null
-                  ? streamData.error!.e as Exception
-                  : Exception(),
-            );
+          final exception = streamData.error?.e != null
+              ? streamData.error!.e as Exception
```
[the remark this answers is not in the document]
Comment (grafovdenis) [reply packages/relation/lib/src/builder/entity_state_builder.dart:96]:
```dart
class ExceptionWrapper {
  const ExceptionWrapper(this.e);

  /// Some exception
  final dynamic e;
}
```

Можно поменять `dynamic` на `Exception`, но мы потеряем возможность пихать туда `Error`.

File: packages/relation/lib/src/builder/entity_state_builder.dart [hunk 7]
```diff
@@ -54,14 +55,17 @@ class EntityStateBuilder<T> extends StatelessWidget {
...
-  /// Child of builder
-  final DataWidgetBuilder<T> child;
+  /// Returns [Widget] depend on [streamedState]'s data
+  final DataWidgetBuilder<T> builder;
 
   /// Loading child of builder
   final DataWidgetBuilder<T>? loadingBuilder;
 
+  /// Error child of builder with previous data
+  final DataWidgetErrorBuilder<T>? errorDataBuilder;
```
[the remark this answers is not in the document]
Comment (IlyaMax) [reply packages/relation/lib/src/builder/entity_state_builder.dart]: лучше тогда поменять название переменной, правильнее dataerror или даже лучше errorwithdata

**Review by dkrutskikh — APPROVED** [review]

**Review by artem-zaitsev — APPROVED** [review]

**Pull request merged**